optuna/samplers/cmaes.py

```diff
@@ -89,6 +90,15 @@ def objective(trial):
 
             Note that the parameters of the first trial in a study are always sampled
             via an independent sampler, so no warning messages are emitted in this case.
+
+        consider_pruned_trials:
```

Comment:
This parameter seems a bit like a compromise as it doesn't behave all too nicely with the `MedianPruner`. I'm thus wondering if we shouldn't provide this feature (parameter) as experimental, and keyword-only. It might not happen in the near future but is it likely that we can improve this somehow, e.g. with learning curve prediction? In that case I imagine we no longer need this parameter.

Reply:
Yes, I totally agree with you. I will make this argument keyword-only, and experimental. I don't know how to make the argument experimental using `optuna._experimental` module, so I just note it to the docstring.

> It might not happen in the near future but is it likely that we can improve this somehow, e.g. with learning curve prediction? In that case I imagine we no longer need this parameter.

Yes, I expect that this argument will be removed in the future. Thanks for clarifying!